Sources/Foundation/FileDescriptorPort.swift

```diff
@@ -0,0 +1,225 @@
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2021 - 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+
+@_implementationOnly import CoreFoundation
+
+#if os(Linux)
+
+    let kCFRunLoopSourceOrder: CFIndex = 600
+
+    public protocol FileDescriptorPortDelegate: PortDelegate {
```

Comment:
The goal of this swift class is to give clients of the API ability to use file descriptors with RunLoop class, since CFRunLoop as API is going away (even tho it's still available on Linux)
From your comment I assume it's not possible to enable this in Darwin due to reasons other than source compatibility, right? Asking because on source code and functional levels CFFileDescriptor on Darwin and this implementation are pretty much the same

Reply:
i just realized: this repository is never merged into apple private code. am i right?

Reply:
Can you perhaps ship this as an external package? Swift-corelibs-foundation seems to be more guarded than linux mainline, it‘s becoming ridiculous.